### PR TITLE
improve .view() performance

### PR DESCRIPTION
**Pull request by ngimel**
Improve .view() performance by not calling set_ and instead restriding returned alias. This improves performance of .view() operation from ~500ns to ~360 ns

Test plan: covered by existing tests

### CODE REVIEW

**Review by albanD — APPROVED** [review]

**Comment by ngimel** [comment]
@alband can you please take another look, I further improved perf (now it's at ~280ns), but PR is changed quite a bit after approval, sorry about that.

File: aten/src/ATen/native/TensorShape.cpp [hunk 1]
```diff
@@ -1201,33 +1227,11 @@ Tensor view(const Tensor& self, IntArrayRef size) {
...
-                                                self.type_set());
-    impl->set_storage_offset(self.storage_offset());
-    impl->set_sizes_and_strides(self.sizes(), self.strides());
-    self_ = Tensor(std::move(impl));
-  }
-#ifdef BUILD_NAMEDTENSOR
-  namedinference::propagate_names(self_, self);
-#endif
-  return self_;
+    return alias_with_sizes_and_strides(self, self.sizes(), self.strides());
```
Comment (albanD) [inline aten/src/ATen/native/TensorShape.cpp]: What is the purpose of the `alias()` function? Is it used in other places?

**Review by albanD — APPROVED** [review]
lgtm

File: aten/src/ATen/native/TensorShape.cpp (hunk 1 above)
Comment (ngimel) [reply aten/src/ATen/native/TensorShape.cpp]: It is used somewhere related to named tensors, I did not dig deep.

Comment (albanD) [reply aten/src/ATen/native/TensorShape.cpp]: Ok, we can keep it then. I was surprised as this is a cpp-only function.

**Pull request closed without merging**